### DISCUSSION TITLE
DSL expression for region expression based on distance

### DISCUSSION THREAD

**Discussion by lungsi** [arbor-sim/arbor#1886] [Q&A]
Hi,
If I wanted to make a label for regions of apical dendrites that are until d distance from the soma, (provided it is possible) I was wondering if anyone can show me how to write its DSL express?

The [documentation](https://docs.arbor-sim.org/en/latest/concepts/labels.html#region-expressions) has an example for [(proximal-interval start:locset)](https://docs.arbor-sim.org/en/latest/concepts/labels.html#region-expressions)
```
(proximal-interval (sum (location 1 0.8) (location 2 0.3)))
```
But, if I understand it correctly **0.8** and **0.3** are distances from the proximal ends of respective branches (1 & 2). Can I set the distance from the soma for all branches restricted to those with the tag for apical dendrite?

**Comment by thorstenhater** [comment]
Hi @lungsi, 

you can compose these expression from atomic to complex. I personally use the arbor-gui tool to see the results
of the change.
See attached screenshot, how an expression for your requirements might look like. Usually we would name the intermediate
results, though.

<img width="1392" alt="Screenshot 2022-05-03 at 16 15 49" src="https://user-images.githubusercontent.com/24411438/166470467-792bb14d-3d33-49b4-b9ba-7c713f945d0c.png">
